api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -876,6 +876,13 @@ message Cluster {
   // DNS resolution configuration which includes the underlying dns resolver addresses and options.
   core.v3.DnsResolutionConfig dns_resolution_config = 53;
 
+  // Optional configuration for having server init block on async DNS resolution. Only applicable for
+  // :ref:`STRICT_DNS<envoy_v3_api_enum_value_config.cluster.v3.Cluster.DiscoveryType.STRICT_DNS>`,
+  // or :ref:`LOGICAL_DNS<envoy_v3_api_enum_value_config.cluster.v3.Cluster.DiscoveryType.LOGICAL_DNS>`.
+  // If true, server startup blocks on async DNS resolution. If false, server will complete
+  // initialization whether or not DNS resolution has completed. Defaults to true.
+  google.protobuf.BoolValue wait_for_dns_on_init = 54;
```

Comment:
open to a different name.

Reply:
+1, I think this seems like a useful general option and we could just have it apply everywhere.

Reply:
Great point. Will update. I can can land this PR for only dns based clusters, and if there is appetite I can do a subsequent one for other cluster types.